### ISSUE TITLE
Add feature for clamping the x/y translation

### ISSUE THREAD

**Issue by scottyantipa** [transcriptic/react-map-interaction#8]
The user of MapInteraction should be able to specify the min and max translation in both the x and y dimensions.

**Comment by c-emil** [comment]
I definitely vote for that feature. This library is just incredible by the way. It's the first library for React that I found that actually supports panning + zooming together with support for mobile, just amazing!

**Comment by c-emil** [comment]
By the way, have you tried using translate3d instead of translate + scale? I think it would be worth it because translate3d uses hardware acceleration.

**Comment by Seb-AS** [comment]
Any idea how to implement the bounds?, I've implemented double click/tap to zoom

**Comment by scottyantipa** [comment]
@Seb-AS I would like to build this functionality into the library, but in the meantime: The `MapInteraction` component takes props.scale and props.translate. So you could implement your bounding logic in the parent component. You'll need to pass `onChange` to MapInteraction, and then handle the { scale, translation } object that it passes back.

Something like:
```
applyBounds(scale, translation) {
  this.setState({ scale: scale, translation: ... do your bounding here ....})
}

render() {
  return (
    <MapInteraction
      translation={this.state.translation}
      scale={this.state.scale}
      onChange={({ scale, translation}) => this.applyBounds(scale, translation)}
    />
  )
}
```

**Comment by scottyantipa** [comment]
I'll take a look at this today.

**Comment by scottyantipa** [comment]
Published in version 1.2.0
